counsel.el

```diff
@@ -1280,14 +1285,18 @@ Like `locate-dominating-file', but DIR defaults to
      t)))
 
 ;;;###autoload
-(defun counsel-git (&optional initial-input)
+(defun counsel-git (&optional initial-input initial-directory)
   "Find file in the current Git repository.
-INITIAL-INPUT can be given as the initial minibuffer input."
+INITIAL-INPUT can be given as the initial minibuffer input.
+INITIAL-DIRECTORY, if non-nil, is used as the root directory for search."
   (interactive)
   (counsel-require-program counsel-git-cmd)
-  (let ((default-directory (counsel-locate-git-root)))
+  (let ((enable-recursive-minibuffers t)
+        (default-directory (or initial-directory
+                               (counsel-locate-git-root))))
```

Comment:
This doesn't seem to work with `counsel-git-action`:

1. `make plain`
2. `M-x counsel-git RET`
3. `C-x C-d doc RET RET`

This results in `doc/doc/Changelog.org` being visited, rather than the expected `doc/Changelog.org`.